javascript/code-challenges/tree/tree.js

```diff
@@ -110,8 +110,29 @@ class BinaryTree {
     };
     _walk(this.root);
     return treeMax;
-  }
-  
+  };
+
+  breadthFirst() {
+    let queue = [];
+    let results = [];
+    let index = 0;
+
+    let _traverse = (node) => {
+    if(!node) return;
+    else if(node) {
+        queue[index] = node;
+        results[index] = node.value;
+      }
+    while(queue.length > 0) {
+      // let current = queue[index];
+      if (node.left) { node.left = queue[index++] }; 
+      if (node.right) { node.right = queue[index++] };
+      delete queue[0]; 
+      _traverse(queue[0]);
+
+      };
+    };
+  };
```

Comment:
Nathan what you have so far looks really great! When you have a chance to come back to this, I will regrade for you!